src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -58,13 +58,90 @@ class PredictionCache {
   std::shared_ptr<metrics::RatioCounter> hit_ratio_;
 };
 
-template <typename Scheduler>
+struct DeadlineCompare {
+  bool operator()(const std::pair<Deadline, PredictTask>& lhs,
+                  const std::pair<Deadline, PredictTask>& rhs) {
+    return lhs.first > rhs.first;
+  }
+};
```

Comment:
Can you add a unit test for this? I tend to mess up my comparisons and I want to be safe here.